modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -261,7 +279,9 @@ struct ReLUFunctor
         if (backendId == DNN_BACKEND_INFERENCE_ENGINE)
             return slope >= 0 || !INF_ENGINE_VER_MAJOR_EQ(INF_ENGINE_RELEASE_2019R1);
 #endif
-        return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE ||
+        return backendId == DNN_BACKEND_OPENCV ||
+               (backendId == DNN_BACKEND_CUDA && haveCUDA()) ||
```

Comment:
It's better to remove `haveCUDA` checks from `supportBackend`. We already checked it at `initCUDABackend` once.